swap/src/protocol/alice/event_loop.rs

```diff
@@ -174,11 +203,76 @@ impl EventLoop {
     }
 
     async fn handle_swap_request(
-        &self,
-        _msg: SwapRequest,
-        _channel: ResponseChannel<SwapResponse>,
-    ) {
+        &mut self,
+        swap_request: SwapRequest,
+        channel: ResponseChannel<SwapResponse>,
+        bob_peer_id: PeerId,
+    ) -> Result<()> {
+        // 1. Check if acceptable request
+        // 2. Send response
+
+        let btc_amount = swap_request.btc_amount;
+        let xmr_amount = btc_amount.as_btc() * RATE as f64;
+        let xmr_amount = monero::Amount::from_monero(xmr_amount)?;
+        let swap_response = SwapResponse { xmr_amount };
+
+        self.swarm
+            .send_swap_response(channel, swap_response)
+            .context("Failed to send swap response")?;
```

Comment:
We were talking about names of this at some point. Really what we are doing here is requesting a quote :)

Reply:
True, let's do that in follow-up